tests/unit/rbtree/main.c

```diff
@@ -245,9 +245,91 @@ void test_rbtree_spam(void)
 	} while (size < MAX_NODES);
 }
 
+/**
+ * @brief Test that remove a node with abnormal color.
+ *
+ * @details Initialize a tree and insert it, and then modify
+ * the node's color manually forcely. And test the remove
+ * operations' behavor.
+ *
+ * @ingroup lib_rbtree_tests
+ *
+ * @see rb_get_min(), rb_get_max(), rb_remove()
+ */
+void test_rb_remove_black_err(void)
+{
+	struct rbtree tree;
+	struct rbnode nodes[5];
+
+	/* Initialize a tree and insert it */
+	(void)memset(&tree, 0, sizeof(tree));
+	tree.lessthan_fn = node_lessthan;
+	(void)memset(nodes, 0, sizeof(nodes));
+	/* Try to get min node, nil will be returned */
+	zassert_true(rb_get_min(&tree) == NULL, "the new tree is not empty");
+
+	for (int i = 0; i < 4; i++) {
+		rb_insert(&tree, &nodes[i]);
+	}
+
+	/* Check if tree's max and min node are our expected */
+	zassert_true(rb_get_min(&tree) == &nodes[0], "the tree is incorrect");
+	zassert_true(rb_get_max(&tree) == &nodes[3], "the tree is incorrect");
+
+	/* Try to remove a node that not in the tree will be returned directly */
+	rb_remove(&tree, &nodes[4]);
+
+	/* Manually set the color of the node to make the tree abnormal,
+	 * so as to satisfy the abnormal branch.
+	 */
+	CHECK(&nodes[3]);
+	uintptr_t *p = (void *) &nodes[3];
+
+	*p = (*p & ~1UL) | (uint8_t)1;
+	rb_remove(&tree, &nodes[2]);
+}
+
+/**
+ * @brief Test that remove a node with abnormal color.
```

Comment:
It better to change it like below.

"Test removing a node with abnormal color."